src/galaxy/StarSystemGenerator.cpp

```diff
@@ -1259,6 +1259,37 @@ void StarSystemRandomGenerator::MakeBinaryPair(SystemBody *a, SystemBody *b, fix
 	b->m_orbMax = orbMax;
 }
 
+SystemBody* StarSystemRandomGenerator::MakeSystemBody(RefCountedPtr<StarSystem::GeneratorAPI> system, const std::string &systemName, const char *bodySuffix)
+{
+	SystemBody *newBody = system->NewBody();
+	newBody->m_name = systemName;
+	if (bodySuffix != nullptr) {
+		newBody->m_name += " ";
+		newBody->m_name += bodySuffix;
+	}
+
+	return newBody;
+}
+
+SystemBody* StarSystemRandomGenerator::MakeGravFor(RefCountedPtr<StarSystem::GeneratorAPI> system, const std::string &systemName, SystemBody * body1, SystemBody * body2)
```

Comment:
Whilst you're refactoring, could you rename this to `MakeGravPointForBodies` please?